pkg/publish/docker.go

```diff
@@ -145,7 +145,21 @@ func Docker(manifest model.Manifest, hub string, tags []string, cosignkey string
 			// Sign images *after* push -- cosign only works against real
 			// repositories (not valid against tarballs)
 			if cosignEnabled {
-				if err := util.VerboseCommand("cosign", "sign", "--key", cosignkey, img.NewReference(arch), "-y").Run(); err != nil {
+				imgRef, err := name.ParseReference(img.NewReference(arch))
+				if err != nil {
+					return fmt.Errorf("failed to parse image reference %v: %v", img.NewReference(arch), err)
+				}
+				newImg, err := remote.Image(imgRef, remote.WithAuthFromKeychain(authn.DefaultKeychain))
```

Comment:
Why do we need to get it from the remote? Aren't we the ones pushing it? or we don't really know since its `docker` pushing I suppose